### PR TITLE
Tests(E2E) : Fix flaky e2e tests on dashboard

### PR DESCRIPTION
**Pull request by MedWassCh**
## Description
A fix of the flaky e2e on dashboard

**Fixes** # MON-22297

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [x] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/tests/e2e/cypress/e2e/Dashboards/05-dashboard-check-permissions/index.ts [hunk 1]
```diff
@@ -9,10 +9,10 @@ import dashboardViewerUser from '../../../fixtures/users/user-dashboard-viewer.j
 
 before(() => {
   cy.startWebContainer();
-  cy.execInContainer({
-    command: `sed -i 's@"dashboard": 0@"dashboard": 3@' /usr/share/centreon/config/features.json`,
-    name: Cypress.env('dockerName')
-  });
+  // cy.execInContainer({
```
Comment (Thebarda) [inline centreon/tests/e2e/cypress/e2e/Dashboards/05-dashboard-check-permissions/index.ts:12]: Is this command relevant?

**Review by Thebarda — APPROVED** [review]

File: centreon/tests/e2e/cypress/e2e/Dashboards/05-dashboard-check-permissions/index.ts (hunk 1 above)
Comment (MedWassCh) [reply centreon/tests/e2e/cypress/e2e/Dashboards/05-dashboard-check-permissions/index.ts]: Yes i already remove the comment

File: centreon/tests/e2e/cypress/e2e/Dashboards/05-dashboard-check-permissions/index.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ import dashboardCreatorUser from '../../../fixtures/users/user-dashboard-creator
 import dashboardViewerUser from '../../../fixtures/users/user-dashboard-viewer.json';
 
 before(() => {
-  cy.startWebContainer();
+  cy.startWebContainer({ version: 'develop' });
```
Comment (kduret) [inline centreon/tests/e2e/cypress/e2e/Dashboards/05-dashboard-check-permissions/index.ts:11]:
```suggestion
  cy.startWebContainer();
```

**Review by kduret — COMMENTED** [review]
Other fails happen in the CI with dashboard tests
Could you remove `cy.reload();` in other tests please ?

File: centreon/tests/e2e/cypress/e2e/Dashboards/commands.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    cy.getByTestId({ testId: 'role-input' }).eq(1).should('be.visible');
+
+    return cy
+      .get('[data-testid="role-input"]')
+      .should('be.visible')
+      .then(($element) => {
+        if ($element.length === 3) {
+          cy.getByTestId({ testId: 'CloseIcon' }).click();
+
+          return cy.wrap(true);
```
Comment (kduret) [inline centreon/tests/e2e/cypress/e2e/Dashboards/commands.ts:17]:
```suggestion
          return true;
```

File: centreon/tests/e2e/cypress/e2e/Dashboards/commands.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+/* eslint-disable @typescript-eslint/no-namespace */
+
+import 'cypress-wait-until';
+
+Cypress.Commands.add('customWaitUntil', (accessRightsTestId) => {
+  const openModalAndCheck: () => Cypress.Chainable<boolean> = () => {
```
Comment (kduret) [inline centreon/tests/e2e/cypress/e2e/Dashboards/commands.ts:6]:
```suggestion
  const openModalAndCheck: () => boolean = () => {
```

File: centreon/tests/e2e/cypress/e2e/Dashboards/commands.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
+/* eslint-disable @typescript-eslint/no-namespace */
+
+import 'cypress-wait-until';
+
+Cypress.Commands.add('customWaitUntil', (accessRightsTestId) => {
```
Comment (kduret) [inline centreon/tests/e2e/cypress/e2e/Dashboards/commands.ts:5]: rename `customWaitUntil` with an appropriate name which explain the intention of the method
```suggestion
Cypress.Commands.add('customWaitUntil', (accessRightsTestId) => {
```

File: centreon/tests/e2e/cypress/e2e/Dashboards/commands.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+          return cy.wrap(true);
+        }
+        cy.getByTestId({ testId: 'CloseIcon' }).click();
+
+        return openModalAndCheck();
+      });
+  };
+
+  return cy.waitUntil(openModalAndCheck, {
+    errorMsg: "L'élément n'est pas devenu visible dans les 30 secondes.",
```
Comment (kduret) [inline centreon/tests/e2e/cypress/e2e/Dashboards/commands.ts:26]: translate to english

File: centreon/tests/e2e/cypress/support/commands.ts [hunk 7]
```diff
@@ -1,9 +1,9 @@
 /* eslint-disable @typescript-eslint/no-namespace */
 
-import 'cypress-wait-until';
```
Comment (kduret) [inline centreon/tests/e2e/cypress/support/commands.ts:3]: should be imported here

File: centreon/tests/e2e/cypress/e2e/Dashboards/commands.ts [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+Cypress.Commands.add('customWaitUntil', (accessRightsTestId) => {
+  const openModalAndCheck: () => Cypress.Chainable<boolean> = () => {
+    cy.getByTestId({ testId: accessRightsTestId }).invoke('show').click();
+    cy.getByTestId({ testId: 'role-input' }).eq(1).should('be.visible');
+
+    return cy
+      .get('[data-testid="role-input"]')
+      .should('be.visible')
+      .then(($element) => {
+        if ($element.length === 3) {
```
Comment (kduret) [inline centreon/tests/e2e/cypress/e2e/Dashboards/commands.ts:14]: `3` should be a parameter of the function

File: centreon/tests/e2e/cypress/e2e/Dashboards/commands.ts (hunk 5 above)
Comment (Thebarda) [reply centreon/tests/e2e/cypress/e2e/Dashboards/commands.ts:5]: I would rename the command `waitUntilForDashboardRoles`

**Review by kduret — APPROVED** [review]

**Pull request merged**